pkg/operatorsource/deleted.go

```diff
@@ -89,6 +94,22 @@ func (r *deletedReconciler) Reconcile(ctx context.Context, in *v1.OperatorSource
 
 	r.logger.Info("Finalizer removed, now garbage collector will clean it up.")
 
+	// Now let's make sure the garbage collector finished its work before we try to recreate it.
```

Comment:
Shouldn't we do this only if it is a default `OperatorSource`?

Reply:
Sorry, this should be held. I'm using this PR as a way of testing against the CI environment.